Serenity-Website/src/MoodBoard/MoodBoard.jsx

```diff
@@ -1,12 +1,127 @@
-import './MoodBoard.css'
+import { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCalendar, faX } from "@fortawesome/free-solid-svg-icons";
+
+import './MoodBoard.css';
+import angry  from '../assets/angry.png';
+import sad from '../assets/sad.png';
+import neutral from '../assets/neutral.png';
+import happy from '../assets/happy.png';
+import excited from '../assets/excited.png';
+import Header from "../Header/Header";
+
+const moods = {
+    // moood object 
+    angry: <img src={angry}/>,
+    sad: <img src={sad} />,
+    neutral: <img src={neutral}/>,
+    happy: <img src={happy}/>,
+    excited: <img src={excited} />
+
+};
 
 function MoodBoard (){
+    // set default mood to null
+    const [selectedMood, setSelectedMood] = useState(null);
+    
+    //set current date
+    const [currentDate, setCurrentDate] = useState('');
+    
+    //state hook to set success message for when user submits a mood.
+    const [successMessage, setSuccessMessage] = useState('');
+    
+    //track if mood has been submitted
+    const [moodSubmitted, setMoodSubmitted] = useState(false);
+
+    // set error message
+    const[errorMessage, setErrorMessage] = useState('');
+
+    const navigate = useNavigate();
+
+    useEffect(() => {
+        // Set the current date and render when the component mounts 
+        const now = new Date();
+        const formattedDate = now.toLocaleString('en-US', 
+            {weekday: 'long', day: 'numeric', month: 'long', hour: 'numeric', minute: 'numeric'});
+        setCurrentDate(formattedDate);
+    }, []);
+
+    const backendUrlAddress = import.meta.env.VITE_BACKEND_ADDRESS
+    const url = `${backendUrlAddress}/moodlog`
+
+    const handleMoodSelect = (mood) => {
+        // handle mood changes
+        setSelectedMood(prevMood => prevMood === mood ? null : mood);
+    }
+
+    const handleMoodSubmit = async() => {
+        if(!selectedMood) {
+            alert('Please select a mood');
+            return;
+        }
+
+        try {
+            const response = await fetch (url, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                credentials : 'include',
+                body: JSON.stringify ({ mood: selectedMood })
```

Comment:
From your data model, it seems like you also plan to POST the userID and DateTime. Are you planning to implement this in a future PR? 

Also - we talked about this last Friday, but it'd be good to change your data models to incorporate an Entry object that ties together the selected mood and all activity categories + selected activity options together. This is something you can change in a following PR.

Reply:
✅